dfastmi/io/RiversObject.py

```diff
@@ -188,36 +169,38 @@ def _read_rivers(self):
             The name of the river configuration file (default "rivers.ini").    
         """
         self.version = "2.0"
-        self._initialize()        
-        self._initialize_advanced()
+        for branch in self.branches:
+            for reach in branch.reaches:
+                self._initialize(river_data, branch, reach)
+                self._initialize_advanced(river_data, branch, reach)       
         
+        self._verify_reaches()
         
-        for ib in range(len(self.branches)):
-            reaches = self.allreaches[ib]
-            branch = self.branches[ib]                
-            for i in range(len(reaches)):
-                reach = reaches[i]
-                
-                hydro_q = self.hydro_q[ib][i]
-                hydro_t = self.hydro_t[ib][i]
-                auto_time = self.autotime[ib][i]
-                qfit = self.qfit[ib][i]
-                self._verify_consistency_HydroQ_and_HydroT(hydro_q, hydro_t, auto_time, qfit, branch, reach)
-
-                use_tide = self.tide[ib][i]
-                tide_boundary_condition = self.tide_bc[ib][i]
-                self._verify_consistency_Hydro_and_TideBC(use_tide, hydro_q, tide_boundary_condition, branch, reach)
+    def _verify_reaches(self):
+        for branch in self.branches:
+            for reach in branch.reaches:
+                hydro_q = reach.hydro_q
+                hydro_t = reach.hydro_t
+                auto_time = reach.autotime
+                qfit = reach.qfit
+                self._verify_consistency_HydroQ_and_HydroT(hydro_q, hydro_t, auto_time, qfit, branch.name, reach.name)
+
+                use_tide = reach.tide
+                tide_boundary_condition = reach.tide_bc
+                self._verify_consistency_Hydro_and_TideBC(use_tide, hydro_q, tide_boundary_condition, branch.name, reach.name)
                 
-                celer_form = self.cform[ib][i]
-                prop_q = self.prop_q[ib][i]
-                prop_c = self.prop_c[ib][i]
-                celer_discharge = self.cdisch[ib][i]    
-                self._verify_CelerForm_with_PropQ_and_PropC(celer_form, prop_q, prop_c, celer_discharge, branch, reach) 
-
-    def _verify_CelerForm_with_PropQ_and_PropC(self, celer_form, prop_q, prop_c, celer_discharge, branch, reach):
+                celer_form = reach.celer_form
+                celer_object = reach.celer_object
+                self._verify_CelerForm_with_PropQ_and_PropC(celer_form, celer_object, branch.name, reach.name)
+
+    def _verify_CelerForm_with_PropQ_and_PropC(self, celer_form, celer_object, branch, reach):
+    # prop_q = reach.prop_q
```

Comment:
whats the plan with this commented part?

Reply:
Is removed